### PR TITLE
Add input validation checks for the Client Requests

### PR DESCRIPTION
**Pull request by ghost**
Add some of the input validation checks described in [#86](https://github.com/frequenz-io/frequenz-service-electricity-trading/issues/86). Note that this checks for only the ones that can be validated in the client (e.g. the `Client` does not have knowledge of which `delivery_area` is valid or not, so it can not check for that).

### CODE REVIEW

**Review by micaebe — APPROVED** [review]
LGTM

File: src/frequenz/client/electricity_trading/_client.py [hunk 1]
```diff
@@ -280,6 +286,60 @@ async def stream_public_trades(
...
+            )
+        if not isinstance(peak_price_delta, _Sentinel) and peak_price_delta is not None:
+            validate_decimal_places(
+                peak_price_delta.amount, PRECISION_DECIMAL_PRICE, "peak price delta"
+            )
+        if not isinstance(display_quantity, _Sentinel) and display_quantity is not None:
+            validate_decimal_places(
+                display_quantity.mwh, PRECISION_DECIMAL_QUANTITY, "display quantity"
+            )
+        if not isinstance(delivery_period, _Sentinel) and delivery_period is not None:
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/client/electricity_trading/_client.py]: The `delivery_period` is not defined as `_Sentinel` type in the function signature. So either the _Sentinel check is redundant here or the function signature is wrong

File: tests/test_client.py [hunk 2]
```diff
@@ -34,6 +34,10 @@
     TradeState,
 )
 
+# Set the year of the delivery starts to one year from now to make sure
+# we never get an expired delivery start
+NEXT_YEAR = datetime.now().year + 1
```
Comment (daniel-zullo-frequenz) [inline tests/test_client.py]: you might want to add timezone information

File: src/frequenz/client/electricity_trading/_client.py (hunk 1 above)
Comment (ghost) [reply src/frequenz/client/electricity_trading/_client.py]: Oh yea good call that's a typo, `delivery_period` should never be `Sentinel`

File: tests/test_client.py (hunk 2 above)
Comment (ghost) [reply tests/test_client.py]: I actually ended up removing the constant as it was only used once, and set the delivery start to the next day and not next year as it is more realistic.

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**